In [102]:
pip install pandas

Note: you may need to restart the kernel to use updated packages.


In [103]:
import pandas as pd

In [104]:
df = pd.read_csv("captions_csv.csv")
df

Sr No      Image File  \
0          1      img/insta1   
1          2      img/insta2   
2          3      img/insta3   
3          4      img/insta4   
4          5      img/insta5   
...      ...             ...   
20510  20511  img/insta20511   
20511  20512  img/insta20512   
20512  20513  img/insta20513   
20513  20514  img/insta20514   
20514  20515  img/insta20515   

                                                 Caption  
0                                                    NaN  
1                                                   bye   
2      Ok, a few more... sorry I just had so much fun...  
3      This was one of my favorite shoots I’ve ever d...  
4                   Wrapped round my finger like a ring   
...                                                  ...  
20510                                        Cowgirl :)   
20511                                                <3   
20512         I love me and Kylie's nail polish colors!   
20513                                             Fammm   
20514                                       Disneyland!   

[20515 rows x 3 columns]

In [105]:
cf = pd.read_csv("captions_csv2.csv")
cf

20516  img2/insta20516  \
0      20517  img2/insta20517   
1      20518  img2/insta20518   
2      20519  img2/insta20519   
3      20520  img2/insta20520   
4      20521  img2/insta20521   
...      ...              ...   
14406  34923  img2/insta34923   
14407  34924  img2/insta34924   
14408  34925  img2/insta34925   
14409  34926  img2/insta34926   
14410  34927  img2/insta34927   

                       wHaT dAy Is It Even #stayhomeclub  
0      Vitamin C for your fashion diet #KatyPursey #S...  
1      do you know the hotter the fire the purer the ...  
2                                       @ryanseacrest 👍🏻  
3               working hard or hardly working idk 🤷🏼‍♀️  
4      Step into your destiny with The Bondie 🔮✨ #Sho...  
...                                                  ...  
14406  Mexican vegan pizza I made from scratch. Meat ...  
14407                    Hummm....Can u spot the error?   
14408                    Snuggling with the dry cleaning  
14409                                  Morning practice   
14410                                  Up up and away!!!  

[14411 rows x 3 columns]

In [106]:
! pip install git+https://github.com/huggingface/transformers.git accelerate datasets

  Cloning https://github.com/huggingface/transformers.git to /private/var/folders/r0/9brhjn911d79nb9n6r_j73t80000gp/T/pip-req-build-px7w5rgq
  Running command git clone --filter=blob:none --quiet https://github.com/huggingface/transformers.git /private/var/folders/r0/9brhjn911d79nb9n6r_j73t80000gp/T/pip-req-build-px7w5rgq
^C
ERROR: Operation cancelled by user


In [ ]:
from datasets import load_dataset

ds = load_dataset("mrSoul7766/instagram_post_captions")

In [ ]:
ds

DatasetDict({
    train: Dataset({
        features: ['image', 'caption'],
        num_rows: 30191
    })
})

In [ ]:
from datasets import load_dataset

da = load_dataset("kkcosmos/instagram-images-with-captions")

In [ ]:
da

DatasetDict({
    train: Dataset({
        features: ['image', 'item_id', 'image_path', 'caption'],
        num_rows: 28360
    })
    test: Dataset({
        features: ['image', 'item_id', 'image_path', 'caption'],
        num_rows: 3152
    })
})

# Instagram Images with Captions Datasets #

โครงการนี้เป็นการสร้างโมเดล Machine Learning ที่ใช้ในการสร้างแคปชั่น (captions) สำหรับรูปภาพที่ผู้ใช้อัปโหลดบน Instagram โดยอาศัยชุดข้อมูล Instagram Images with Captions Datasets ที่มีอยู่เป็นฐานข้อมูลหลัก

การทำโมเดลนี้มีวัตถุประสงค์เพื่อช่วยให้ผู้ใช้งานอินเทอร์เน็ตที่ต้องการอัปโหลดรูปภาพลง Instagram สามารถสร้างแคปชั่นได้โดยง่ายและมีคุณภาพ โดยทำให้เพิ่มประสิทธิภาพในการจัดการเนื้อหาและเพิ่มความน่าสนใจให้กับรูปภาพที่ถูกโพสต์

    

1. <u> จัดทำโดย </u>
</br>
</br>
6610402175 ฟาร์ฮาน่า มาเหล็ม หมู่ 1
</br>
6610405930 เบญญภา ประคุณหังสิต หมู่ 1
</br>
6610405972 พัทธนันท์ ขันตี หมู่ 1

2. แหล่งที่มา (1) : https://www.kaggle.com/datasets/prithvijaunjale/instagram-images-with-captions?resource=download
</br>
แหล่งที่มา (2) : https://huggingface.co/datasets/mrSoul7766/instagram_post_captions?source=post_page-----7f9b9f938794--------------------------------
</br>
แหล่งที่มา (3) : https://huggingface.co/datasets/kkcosmos/instagram-images-with-captions

3. ข้อมูลชุดนี้จัดเก็บมาจากผู้ใช้ในแอปพลิเคชันอินสตาแกรม

4. ชุดข้อมูล Instagram Images with Captions จัดเก็บในรูปแบบดังนี้

|Format|Binary/Text| Human-Readable|
|-----|----|----|
|CSV|Binary and Text|images(binary) = No, texts = Yes|

5. ชุดข้อมูล Instagram Images with Captions Datasets โดยข้อมูลทั้งหมดมีจำนวน 96629 แคปชั่น

| แหล่งข้อมูล | จำนวนข้อมูล|
|----------|----------|
| https://www.kaggle.com/datasets/prithvijaunjale/instagram-images-with-captions?resource=download | 34926|
| https://huggingface.co/datasets/mrSoul7766/instagram_post_captions?source=post_page-----7f9b9f938794--------------------------------| 30191|
|https://huggingface.co/datasets/kkcosmos/instagram-images-with-captions | 31512|


6. สิ่งที่ต้องการทำนายจากการศึกษาชุดข้อมูลนี้ คือ การทำนายแคปชั่นเพื่อโพสต์ลงโซเชียลมีเดีย โดยวิเคราะห์จากข้อมูลที่ศึกษาได้ในชุดข้อมูล

7. สิ่งใช้ในการทำนาย คือ รูปภาพในชุดข้อมูลและแคปชั่นที่เกี่ยวข้องกับรูปภาพนั้นๆ

8. ในการจัดเตรียมจะมีดังนี้
</br>

- รวมชุดข้อมูลจากทั้งหมดเข้าด้วยกันไว้ในไฟล์เดียว
</br>

- การเตรียมภาพและข้อมูล: การประมวลผลภาพเพื่อให้เหมาะสมกับการนำเข้าโมเดล


9. ข้อมูลที่ไม่น่าจะได้ใช้ในระบบต้นแบบ คือ คอลัมน์ Sr No 

10. ข้อมูลชุดนี้ไม่ควรนำไปจัดเก็บในรูปแบบอื่นเนื่องจากการจัดเก็บข้อมูลในรูปแบบนี้จะทำให้ง่ายต่อการอ่านและสรุปข้อมูล

11. ประเด็นอื่นๆ ที่เกี่ยวข้องกับข้อมูล : เนื่องจากมีข้อมูลรูปภาพที่สามารถระบุถึงบุคคลได้ จึงจำเป็นต้องตระหนักถึงจริยธรรมในการใช้ข้อมูลเพื่อนำไปศึกษา, จากชุดข้อมูลเราไม่สามารถทราบได้ว่ารูปภาพนั้นได้รับการยินยอมให้นำมาใช้เพื่อการศึกษาหรือไม่

12. การมีส่วนร่วมของสมาชิกในกลุ่ม 
</br>


6610402175 ฟาร์ฮาน่า มาเหล็ม หมู่ 1
- ค้นหาข้อมูลและวิเคราะห์

6610405930 เบญญภา ประคุณหังสิต หมู่ 1
- ค้นหาข้อมูลและวิเคราะห์

6610405972 พัทธนันท์ ขันตี หมู่ 1
- จัดทำ เรียบเรียง และสรุปข้อมูล


13. การเปิดเผยการใช้เครื่องมือปัญญาประดิษฐ์ : ใช้ ChatGPT ในการเรียบเรียงข้อมูล